app/services/setting_registry.rb

```diff
@@ -0,0 +1,96 @@
+class SettingRegistry
+  include Singleton
+
+  def initialize
+    @settings = {}
+    load
+  end
+
+  def logger
+    Foreman::Logging.logger('app')
+  end
+
+  def find(name)
+    @settings[name.to_s]
+  end
+
+  # Returns a setting effective value
+  # Call as `Foreman.settings[<name>]`
+  #
+  #   Foreman.settings['default_locale'] => nil.
+  #   Foreman.settings[:default_locale] => nil.
+  #
+  def [](name)
+    find(name)&.value
+  end
+
+  def update(name, value)
+    definition = find(name)
+    raise "Setting definition for #{name} not found, can not set" unless definition
+    db_record = _find_or_new_db_record(name)
+    db_record.value = value
+    db_record.save!
+    definition.load_db_value(db_record)
```

Comment:
and then this becomes 
```suggestion
    definition.value = db_record.value
```